pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -0,0 +1,159 @@
+package notifier
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/grafana/grafana/pkg/infra/log"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	amclient "github.com/prometheus/alertmanager/api/v2/client"
+)
+
+type externalAlertmanager struct {
+	log           log.Logger
+	url           string
+	tenantID      string
+	orgID         int64
+	amClient      *amclient.AlertmanagerAPI
```

Comment:
We need two clients here:
- The Prometheus client handles requests made to the `v2` API
- If we want to make requests to the `v1` API (i.e. update the configuration) we need to manually do so with an HTTP client